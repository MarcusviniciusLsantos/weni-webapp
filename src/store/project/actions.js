import { captureException } from '@sentry/browser';
import projects from '../../api/projects';
import i18n from '../../utils/plugins/i18n';
import { unnnicCallAlert } from '@weni/unnnic-system';
import { fetchFlowOrganization } from '../org/actions';

export default {
  getCurrentProject() {
    const object = window.localStorage.getItem('project');
    if (object) return JSON.parse(object);
    return null;
  },

  getProject(store, { uuid }) {
    return projects.getProject({ uuid });
  },

  getProjects(store, { orgId, page = 1, limit = 20, ordering }) {
    const offset = limit * (page - 1);
    return projects.list(orgId, offset, limit, ordering);
  },

  async createProjectForOrg(
    {
      commit,
      rootState: {
        Org: {
          currentOrg: { uuid },
        },
      },
    },
    { project, onBack, onAccess },
  ) {
    commit('PROJECT_CREATE_REQUEST');
    try {
      let response = null;

      response = await projects.createReadyMadeProject(
        project.name,
        uuid,
        project.dateFormat,
        project.timeZone,
        project.format,
        project.globals,
      );

      let flowOrganization = response.data.flow_organization;

      if (!flowOrganization) {
        flowOrganization = await fetchFlowOrganization(response.data.uuid);
      }

      commit('PROJECT_CREATE_SUCCESS', {
        ...response.data,
        flow_organization: flowOrganization,
      });

      commit('OPEN_MODAL', {
        type: 'confirm',
        data: {
          icon: 'check-circle-1-1',
          scheme: 'feedback-green',
          title: i18n.t('projects.create.confirm_title'),
          description: i18n.t('projects.create.confirm_subtitle'),
          cancelText: i18n.t('projects.create.view_projects'),
          confirmText: i18n.t('projects.create.go_to_project'),
          onClose: onBack,
          onConfirm: (justClose) => {
            justClose();
            onAccess(response.data.uuid);
          },
        },
      });
    } catch (e) {
      commit('PROJECT_CREATE_ERROR', e);
      unnnicCallAlert({
        props: {
          text: i18n.t('projects.create.error'),
          title: 'Error',
          icon: 'check-circle-1-1',
          scheme: 'feedback-red',
          position: 'bottom-right',
          closeText: i18n.t('close'),
        },
        seconds: 3,
      });

      captureException(new Error('PROJECT_CREATE', { cause: e }));
    }
  },

  async changeReadyMadeProjectProperties(
    { getters, commit },
    { projectUuid, first_access },
  ) {
    const { data } = await projects.changeReadyMadeProjectProperties({
      projectUuid,
      first_access,
    });

    commit('setCurrentProject', {
      ...getters.currentProject,
      first_access: data.first_access,
    });
  },

  editProject(store, { name, organization, projectUuid, timezone }) {
    return projects.editProject(name, organization, projectUuid, timezone);
  },

  deleteProject(store, { uuid }) {
    return projects.deleteProject(uuid);
  },

  setCurrentProject(
    { commit },
    {
      uuid,
      name,
      menu = {
        chat: [],
        flows: '',
        inteligence: '',
      },
      organization = {
        uuid: '',
      },
      flow_organization = {
        uuid: '',
        id: '',
      },
      first_access,
      flow_uuid,
      flow_count,
      project_type,
      authorization,
    } = {},
  ) {
    commit('setCurrentProject', {
      uuid,
      name,
      menu,
      organization,
      flow_organization,
      first_access,
      flow_uuid,
      flow_count,
      project_type,
      redirect_url: '',
      authorization,
    });
  },

  clearCurrentProject({ commit }) {
    commit('setCurrentProject', null);
  },

  async createOrUpdateProjectAuthorization(
    store,
    { email, projectUuid, role, chatRole, hasChat },
  ) {
    return projects.createProjectAuthorization({
      email,
      projectUuid,
      role,
      chatRole,
      hasChat,
    });
  },

  async removeProjectAuthorization(store, { email, projectUuid }) {
    return projects.deleteProjectAuthorization({
      email,
      projectUuid,
    });
  },

  async getSuccessOrgStatusByFlowUuid({ state, commit }, { flowUuid, force }) {
    try {
      if (!state.championChatbots[flowUuid] || force) {
        const { has_ia, has_flows, has_channel, has_msg } =
          await projects.apiFlowsGetSuccessOrg({ flowUuid });

        commit('setChampionChatbot', {
          flowUuid,
          has_ia,
          has_flows,
          has_channel,
          has_msg,
        });
      }

      return state.championChatbots[flowUuid];
    } catch (error) {
      commit('setChampionChatbot', {
        flowUuid,
        error: true,
      });

      throw error;
    }
  },
};
