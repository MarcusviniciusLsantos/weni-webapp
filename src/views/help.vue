<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <div class="help-container">
    <div class="header">
      <div class="title">{{ $t('faq.title') }}</div>

      <div class="description">
        <span v-html="$t('faq.description', { phone })"></span>
        <Emoji name="Winking Face" />
      </div>
    </div>

    <div class="content">
      <div :style="{ height: 0 }">
        <UnnnicAccordion
          v-for="(question, index) in questions"
          v-model:open="opens[question.key]"
          :key="index"
          :title="question.title"
          class="question"
        >
          <template v-if="question.video" #actions>
            <UnnnicButton
              @click.stop="openVideo(question.video)"
              type="secondary"
              size="small"
              iconLeft="button-play-1"
            >
              {{ $t('faq.questions.watch_video') }}
            </UnnnicButton>
          </template>

          <span v-html="question.content" />
        </UnnnicAccordion>
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import Emoji from '../components/Emoji.vue';

export default {
  components: {
    Emoji,
  },

  data() {
    return {
      phone: '+ 55 (82) 3022-5978',

      opens: {
        will_the_price_change: false,
        did_push_and_bothub_change_name: false,
        how_can_i_access_the_new_push_and_bothub: false,
        will_my_chatbot_suffer_any_impact_with_this_change: false,
        will_any_functionality_change: false,
        what_will_i_find_about_new_benefits: false,
        is_it_possible_i_lose_any_data_from_my_chatbot: false,
        will_my_integrations_be_affected: false,
        do_my_permissions_remain_the_same: false,
        how_and_who_do_i_ask_for_help: false,
      },
    };
  },

  computed: {
    questions() {
      return [
        {
          key: 'will_the_price_change',
        },
        {
          key: 'did_push_and_bothub_change_name',
          hasVideo: true,
        },
        {
          key: 'how_can_i_access_the_new_push_and_bothub',
          hasVideo: true,
        },
        {
          key: 'will_my_chatbot_be_impacted_by_this_change',
        },
        {
          key: 'will_any_functionality_change',
          hasVideo: true,
        },
        {
          key: 'what_will_i_find_about_new_benefits',
          hasVideo: true,
        },
        {
          key: 'is_it_possible_i_lose_any_data_from_my_chatbot',
        },
        {
          key: 'will_my_integrations_be_affected',
        },
        {
          key: 'do_my_permissions_remain_the_same',
          hasVideo: true,
        },
        {
          key: 'how_and_who_do_i_ask_for_help',
          content: ``,
        },
      ].map((question) => ({
        ...question,
        title: this.$t(`faq.questions.${question.key}.title`),
        content: this.$t(`faq.questions.${question.key}.answer`, {
          phone: this.phone,
        }),
        video: question.hasVideo
          ? this.$t(`faq.questions.${question.key}.video`)
          : null,
      }));
    },
  },

  methods: {
    ...mapActions(['openModal']),

    openVideo(video) {
      this.openModal({
        type: 'youtube-video',
        data: {
          url: video,
        },
      });
    },
  },
};
</script>

<style lang="scss" scoped>
.help-container {
  padding: $unnnic-spacing-inset-md;
  height: 100vh;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;

  .header {
    padding: $unnnic-inset-lg;
    margin-bottom: $unnnic-spacing-stack-sm;
    background-color: $unnnic-color-background-sky;
    border-radius: $unnnic-border-radius-md;
    background-image: url('../assets/banner/Background-FAQ.svg');
    background-repeat: no-repeat;
    background-size: auto 100%;

    .title {
      font-family: $unnnic-font-family-primary;
      font-weight: $unnnic-font-weight-regular;
      font-size: $unnnic-font-size-title-md;
      line-height: $unnnic-font-size-title-md + $unnnic-line-height-md;
      color: $unnnic-color-neutral-black;

      margin-bottom: $unnnic-spacing-stack-xs;
    }

    .description {
      font-family: $unnnic-font-family-secondary;
      font-weight: $unnnic-font-weight-regular;
      font-size: $unnnic-font-size-body-lg;
      line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
      color: $unnnic-color-neutral-cloudy;

      :deep(a) {
        font-weight: bold;
        color: inherit;
      }
    }
  }

  .content {
    $scroll-size: $unnnic-inline-nano;

    flex: 1;
    overflow: overlay;
    min-height: 8rem;

    padding-right: calc(#{$unnnic-inline-xs} + #{$scroll-size});
    width: 100%;

    :deep(a) {
      color: inherit;
    }

    .question:not(:last-child) {
      margin-bottom: $unnnic-spacing-stack-sm;
    }

    &::-webkit-scrollbar {
      width: $scroll-size;
    }

    &::-webkit-scrollbar-thumb {
      background: $unnnic-color-neutral-clean;
      border-radius: $unnnic-border-radius-pill;
    }

    &::-webkit-scrollbar-track {
      background: $unnnic-color-neutral-soft;
      border-radius: $unnnic-border-radius-pill;
    }
  }
}
</style>
