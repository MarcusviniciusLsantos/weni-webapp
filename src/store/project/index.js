import { reactive } from 'vue';
import projects from '../../api/projects';
import actions from './actions';
import mutations from './mutations';

const recentActivities = {};

const state = {
  currentProject: null,
  loadingCreateProject: false,
  errorCreateProject: null,
  templates: {
    status: null,
    data: [],
  },
  championChatbots: {},
  projects: [],
  recentActivities: new Proxy(recentActivities, {
    get(target, name, receiver) {
      if (!Reflect.has(target, name)) {
        const recentActivities = reactive({
          status: null,
          next: null,
          data: [],
        });

        recentActivities.loadNext = async () => {
          recentActivities.status = 'loading';

          await projects
            .latestActivities({
              projectUuid: name,
              limit: 20,
              next: recentActivities.next,
            })
            .then(({ data }) => {
              if (data.next) {
                const url = new URL(data.next);
                const cursor = url.searchParams.get('cursor');
                recentActivities.next = cursor;
              }

              const results = [...recentActivities.data, ...data.results];

              recentActivities.data = results.filter(
                (value, index, self) =>
                  index ===
                  self.findIndex(
                    (item) =>
                      item.user === value.user &&
                      item.created_at === value.created_at,
                  ),
              );

              recentActivities.status = data.next == null ? 'complete' : null;
            })
            .catch(() => {
              console.log('error');
              if (recentActivities.status !== 'complete') {
                recentActivities.status = 'error';
              }
            });
        };

        Reflect.set(target, name, recentActivities, receiver);
      }

      return Reflect.get(target, name, receiver);
    },
  }),
};

const getters = {
  currentProject(state) {
    return state.currentProject;
  },
};

export default {
  state,
  getters,
  actions,
  mutations,
};
