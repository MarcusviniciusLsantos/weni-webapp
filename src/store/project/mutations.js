import Vue from 'vue';
import store from '../../store';

export default {
  setCurrentProject: (state, project) => (state.currentProject = project),

  setChampionChatbot: (state, value) => {
    Vue.set(state, 'championChatbots', {
      ...state.setChampionChatbots,
      [value.flowUuid]: value,
    });
  },

  PROJECT_CREATE_REQUEST: (state) => (state.loadingCreateProject = true),
  PROJECT_CREATE_SUCCESS: (state, project) => {
    const projects = store.state.Project.projects.find(
      ({ orgUuid }) => orgUuid === project.organization,
    );

    if (projects) {
      projects.data.push(project);
    }

    state.currentProject = project;
    state.loadingCreateProject = false;
  },
  PROJECT_CREATE_ERROR: (state, createProjectError) => {
    state.errorCreateProject = createProjectError;
    state.loadingCreateProject = false;
  },
};
