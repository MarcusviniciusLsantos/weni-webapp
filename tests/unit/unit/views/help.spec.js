import { vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import help from '@/views/help.vue';
import { org } from '../../__mocks__';
import profile from '../../__mocks__/profile';
import project from '../../__mocks__/project';

vi.mock('@/api/request.js', () => {});

describe('help.vue', () => {
  let wrapper;
  let store;
  let actions;
  let getters;
  let state;

  beforeEach(() => {
    getters = {
      currentOrg: () => {
        return org;
      },
      currentProject: () => {
        return project;
      },
    };
    state = {
      Account: {
        profile,
      },
    };
    actions = {
      updateProfile: vi.fn(),
      updateProfilePicture: vi.fn(),
      removeProfilePicture: vi.fn(),
      openModal: vi.fn(),
    };

    store = createStore({
      getters,
      actions,
      state,
    });

    wrapper = shallowMount(help, {
      global: {
        plugins: [store],
        stubs: {
          UnnnicButton: true,
          emoji: true,
          UnnnicAccordion: true,
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
