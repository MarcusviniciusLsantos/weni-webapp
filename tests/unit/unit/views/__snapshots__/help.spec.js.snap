// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`help.vue > should be rendered properly 1`] = `
<div
  class="help-container"
  data-v-7f8a6a5c=""
>
  <div
    class="header"
    data-v-7f8a6a5c=""
  >
    <div
      class="title"
      data-v-7f8a6a5c=""
    >
      some specific text
    </div>
    <div
      class="description"
      data-v-7f8a6a5c=""
    >
      <span
        data-v-7f8a6a5c=""
      >
        some specific text
      </span>
      <img
        data-v-736c95e5=""
        data-v-7f8a6a5c=""
        src="/src/assets/emoji/u1F609.png"
        title="Winking Face"
      />
    </div>
  </div>
  <div
    class="content"
    data-v-7f8a6a5c=""
  >
    <div
      data-v-7f8a6a5c=""
      style="height: 0px;"
    >
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <!---->
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <unnnicbutton
          data-v-7f8a6a5c=""
          iconleft="button-play-1"
          size="small"
          slot="actions"
          type="secondary"
        >
           some specific text 
        </unnnicbutton>
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <unnnicbutton
          data-v-7f8a6a5c=""
          iconleft="button-play-1"
          size="small"
          slot="actions"
          type="secondary"
        >
           some specific text 
        </unnnicbutton>
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <!---->
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <unnnicbutton
          data-v-7f8a6a5c=""
          iconleft="button-play-1"
          size="small"
          slot="actions"
          type="secondary"
        >
           some specific text 
        </unnnicbutton>
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <unnnicbutton
          data-v-7f8a6a5c=""
          iconleft="button-play-1"
          size="small"
          slot="actions"
          type="secondary"
        >
           some specific text 
        </unnnicbutton>
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <!---->
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <!---->
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <unnnicbutton
          data-v-7f8a6a5c=""
          iconleft="button-play-1"
          size="small"
          slot="actions"
          type="secondary"
        >
           some specific text 
        </unnnicbutton>
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
      <unnnicaccordion
        class="question"
        data-v-7f8a6a5c=""
        title="some specific text"
      >
        <!---->
        <dynamic-stub
          data-v-7f8a6a5c=""
          template="<span>some specific text</span>"
        />
      </unnnicaccordion>
    </div>
  </div>
</div>
`;
